src/clr-core/modal/modal.element.scss

```diff
@@ -0,0 +1,103 @@
+@import './../styles/tokens/generated/index';
+@import './../styles/mixins/utils';
+@import './../styles/utils/close';
+
+:host {
+  --backdrop-opacity: 0.85;
+  --backdrop-color: #{$cds-token-color-neutral-900};
+  --box-shadow-color: #{$cds-token-color-action-500};
+  --border-radius: #{$cds-token-global-border-radius};
+  --border-color: #{$cds-token-color-action-600};
+  --background-color: #{$cds-token-color-neutral-0};
+  --close-icon-color: #{$cds-token-color-neutral-600};
+  --close-icon-color-hover: #{$cds-token-color-neutral-700};
+  --content-box-shadow-color: #{$cds-token-color-neutral-1000};
+  --width: calc(8 * #{$cds-token-space-size-13});
+}
+
+:host([size='sm']) {
+  --width: calc(4 * #{$cds-token-space-size-13});
+}
+
+:host([size='lg']) {
+  --width: calc(12 * #{$cds-token-space-size-13});
+}
+
+:host([size='xl']) {
+  --width: calc(16 * #{$cds-token-space-size-13});
+}
+
+.private-host {
+  position: fixed;
+  top: 0;
+  bottom: 0;
+  right: 0;
+  left: 0;
+  z-index: 1050; // TODO: restore using map-get to get the z-index
+}
+
+button.close {
+  font-size: $cds-token-space-size-8;
+  line-height: $cds-token-space-size-8;
+
+  &:focus,
+  &:hover,
+  &:active {
+    --close-icon-color: var(--close-icon-color-hover);
+  }
+
+  cds-icon {
+    fill: var(--close-icon-color);
+    // per measurement, this results in an icon that is 16x16...
+    @include equilateral($cds-token-space-size-8);
+  }
+}
+
+.modal-dialog {
+  position: relative;
+  z-index: 1050; // TODO: restore using map-get to get the z-index
+  width: var(--width);
+  max-width: 100%;
+}
+
+.modal-content {
+  $modal-boxshadow-y: $cds-token-space-size-1;
+  $modal-boxshadow-spread: $cds-token-space-size-2;
+
+  background-color: var(--background-color);
+  border-radius: var(--border-radius);
+  box-shadow: 0 $modal-boxshadow-y $modal-boxshadow-spread $modal-boxshadow-spread
+    rgba(var(--content-box-shadow-color), 0.2);
```

Comment:
My experience with this pattern has not been a positive one `rgba(var(--content-box-shadow-color), 0.2)`. You may want to verify it in Firefox and Edge...

Reply:
Thank you for pointing this out. Updated how we do this per discussion.